### PR TITLE
Converted to keras core external attention

### PR DESCRIPTION
**Pull request by anas-rz**
Update imports
Remove TensorFlow Addons Dependency
Convert TensorFlow ops in functional component to keras_core ops

Attached git-diff
[Colab Notebook](https://colab.research.google.com/drive/1jczppKZwd29O-2t1vL78IDnBHMrC7k54?usp=sharing)
[eanet_keras_core.diff.txt](https://github.com/keras-team/keras-core/files/12083136/eanet_keras_core.diff.txt)

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/keras_io/tensorflow/vision/eanet.py [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+
+
+class PatchExtract(layers.Layer):
+    def __init__(self, patch_size, **kwargs):
+        super().__init__(**kwargs)
+        self.patch_size = patch_size
+
+    def call(self, images):
+        batch_size = tf.shape(images)[0]
+        patches = tf.image.extract_patches(
```
[the remark this answers is not in the document]
Comment (anas-rz) [reply examples/keras_io/tensorflow/vision/eanet.py:103]: I can create a PR for it. 
Another workaround: For this example can I use frameworks like einops for creating patches?

[the remark this answers is not in the document]
Comment (fchollet) [reply examples/keras_io/tensorflow/vision/eanet.py:103]: Prefer using a conv.

**Comment by fchollet** [comment]
Your way is fine. Conv is likely faster on GPU though. It could be worth trying both if we add the op to the API (just for this example we can keep your way since it's concise).

> Is there any 'do nothing' filter for convolution that can be used for upsampling?

"Do nothing" is just an identity kernel. There is no "sum" operation, it's a matmul. It turns into a sum if your kernel is all ones.

**Comment by anas-rz** [comment]
Here's the initial 'code' version I have got for extracting patches and it is working. Thank you for guidance. Will create a PR for adding this to cross-backend ops.

```import numpy as np
import tensorflow as tf
input_image = tf.random.uniform((1, 224, 224, 3))
patch_h, patch_w = 14, 14
channels_in = 3
out_dim = 588
kernel = tf.eye(patch_h * patch_w * channels_in, out_dim)
kernel = tf.reshape(kernel, (14, 14, 3, 588))
out_conv_layer = tf.nn.conv2d(input_image, kernel, 14, 'VALID')
out_tf = tf.image.extract_patches(input_image, [1, 14, 14, 1], [1, 14, 14, 1], [1, 1, 1, 1], "VALID")
check = tf.equal(out_conv_layer, out_tf)
print(np.unique(check))```

**Comment by fchollet** [comment]
Thanks! `extract_patches` will be a great addition to the `ops.image` API.

Should we wait for this feature before revisiting this example?

**Comment by anas-rz** [comment]
Sure. That'll be better.

**Pull request merged**